src/graph/unit_graph.cc

```diff
@@ -1316,6 +1326,15 @@ void UnitGraph::UnpinMemory_() {
     this->coo_->UnpinMemory_();
 }
 
+void UnitGraph::RecordStream(DGLStreamHandle stream) const {
+  if (this->in_csr_->defined())
+    this->in_csr_->RecordStream(stream);
+  if (this->out_csr_->defined())
+    this->out_csr_->RecordStream(stream);
+  if (this->coo_->defined())
+    this->coo_->RecordStream(stream);
+}
```

Comment:
Do we need anything extra to handle the case where new formats are being created when one format is recorded on a non-default stream?

Reply:
Can we assume `record_stream` is only used in data loading?

Reply:
I could envision it being used in forward/backward where we have parallel layers (.e.g, on heterogeneous graphs). If its not allowed outside of dataloading, we would need to add some check that prevent users from doing it, or at least warn them.

Reply:
Calling `RecordStream` on newly created formats now.